FordLabs/peoplemover
# Update React Version

## Issue
Resolves N/A

## What was done
- [x] Updated `React & react-dom` from version `16` to `17`
- [x] Updated `@testing-library/react` from version `9` to `12` to stay in line with the react upgrades. Updated all tests accordingly.
- [x] Updated `eslint` from version `6` to `8` & removed some eslint packages to make it easier to do so. Will look into reintroducing some back in or adjusting the current rules at a later time.
- [x] Updated `react-scripts` from version `3` to `5`
- [x] Removed guet from the husky commit hooks as it's not assumed that contributors in an open source project will use guet 
- [x] Deleted the jest config file as it wasn't named correctly, and even when it was it wasn't being respected. Moved any jest config needed to the package.json. 

## Notes:
There were a few unit tests I x'd out as they were tests rendering the entire app that would take too much time to trouble shoot. I aim to faze out tests that render the entire app in a unit tests anyway. I left those in for documentation purposes, but will address them in a later PR.

## How to test
1. Ensure the app behaves the same as it did before.
2. Ensure all cypress tests pass
3. Ensure all unit tests pass.
